### PR TITLE
Chapter13:User micropost

### PR DESCRIPTION
**Pull request by ngyihkai**
Ready

### CODE REVIEW

File: app/assets/stylesheets/microposts.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Microposts controller here.
```
Comment (huongnt-2545) [inline app/assets/stylesheets/microposts.scss]: delete this file

File: app/helpers/microposts_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module MicropostsHelper
```
Comment (huongnt-2545) [inline app/helpers/microposts_helper.rb]: delete this file

File: app/views/static_pages/home.html.erb [hunk 3]
```diff
@@ -7,3 +7,22 @@
...
+            <%= render 'shared/micropost_form' %>
+         </section>
+      </aside>
+
+      <div class="col-md-8">
+         <h3>Micropost Feed</h3>
+         <%= render "shared/feed" %>
+      </div>
+   </div>
+<% else %>
```
Comment (huongnt-2545) [inline app/views/static_pages/home.html.erb]: you seem missing some codes in else condition, please check the tutorial

File: db/seeds.rb [hunk 4]
```diff
@@ -22,6 +22,12 @@
              activated: true,
              activated_at: Time.zone.now)
 
+users = User.order(:created_at).take(6)
+30.times do
+ content = Faker::Lorem.sentence(word_count: 5)
```
Comment (huongnt-2545) [inline db/seeds.rb]: you can try replace `faker` to `ffaker` instead
refs: https://github.com/ffaker/ffaker
- Add to Gemfile:
```
group :development, :test do
  # some gems
  gem 'ffaker'
end
```
- At here: 
```suggestion
 content = FFaker::Lorem.sentence(word_count: 5)
```

File: app/views/static_pages/home.html.erb (hunk 3 above)
Comment (ngyihkai) [reply app/views/static_pages/home.html.erb]: i just checked in slide, but i cannot find which part i had skipped.

**Comment by ngyihkai** [comment]
Ready

File: app/assets/stylesheets/microposts.scss (hunk 1 above)
Comment (ghost) [reply app/assets/stylesheets/microposts.scss]: not fix yet

File: app/controllers/static_pages_controller.rb [hunk 5]
```diff
@@ -1,5 +1,17 @@
...
 
-  def help; end
+  def help
+  end
+
+  def about
+  end
+
+  def contact
+  end
```
Comment (ghost) [inline app/controllers/static_pages_controller.rb]:
```suggestion
  def help; end

  def about; end

  def contact; end
```

File: app/helpers/microposts_helper.rb (hunk 2 above)
Comment (ghost) [reply app/helpers/microposts_helper.rb]: not fix yet

File: test/controllers/microposts_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class MicropostsControllerTest < ActionDispatch::IntegrationTest
```
Comment (ghost) [inline test/controllers/microposts_controller_test.rb]: delete this file please

File: test/fixtures/microposts.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+one:
+  content: MyText
+  user: one
```
Comment (ghost) [inline test/fixtures/microposts.yml]: delete this file please

File: test/models/micropost_test.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
```
Comment (ghost) [inline test/models/micropost_test.rb]: delete this file please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ngyihkai** [comment]
Ready

File: app/assets/stylesheets/custom.scss [hunk 9]
```diff
@@ -127,3 +127,51 @@ input.btn.btn-primary {
...
+}
+
+.microposts {
+  .content {
+    display: block;
+    margin-left: 60px;
+    img {
+      display: block;
+      padding: 5px 0;
+  }
```
Comment (ghost) [inline app/assets/stylesheets/custom.scss:170]: align this with img

File: app/controllers/microposts_controller.rb [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    end
+  end
+
+  def destroy
+    if @micropost.destroy
+      flash[:success] = t(:Micropost_deleted)
+    else
+      flash[:danger] = t(:Deleted_fail)
+    end
+      redirect_to request.referer || root_url
```
Comment (ghost) [inline app/controllers/microposts_controller.rb]: align this line with end, please

**Review by ghost — COMMENTED** [review]
small cmt

**Comment by ngyihkai** [comment]
Ready

File: app/assets/stylesheets/custom.scss (hunk 9 above)
Comment (ghost) [reply app/assets/stylesheets/custom.scss:170]: not fix yet

**Review by ghost — COMMENTED** [review]
small comment

**Comment by ngyihkai** [comment]
Ready

**Review by huongnt-2545 — COMMENTED** [review]
Your pull request has 2 commits, please combine in 1

**Comment by ngyihkai** [comment]
Ready

**Review by huongnt-2545 — APPROVED** [review]
merged
you can merge this pull

**Pull request merged**